pymongo/cursor.py

```diff
@@ -684,7 +687,7 @@ def __getitem__(self, index: int) -> _DocumentType:
     def __getitem__(self, index: slice) -> "Cursor[_DocumentType]":
         ...
 
-    def __getitem__(self, index):
+    def __getitem__(self, index: Union[int, slice]) -> Union[_DocumentType, Cursor[_DocumentType]]:
```

Comment:
Are you sure we need to add type hints here? The examples in the docs only add hints to the `@overload` lines, for example: https://docs.python.org/3/library/typing.html#typing.overload

Reply:
Uh good question, I added it because I've been making sure that everything is typed by running mypy with the `--disallow-untyped-defs` flag and mypy will complain that its an untyped def. I can remove it if that is preferred.